### PR TITLE
Remove tslint when pxblue eslintrc is present

### LINKED ISSUE #38: Should not have both .eslintrc.js and tslint.json generated
#### Describe the desired behavior

When generating a new angular project using pxblue linting, the tslint.json should not show up in the root directory

#### Describe the current behavior

It both installed pxblue/eslint-config and generated a tslint.json in the project files.

### PR DESCRIPTION
**Pull request by huayunh**
Fixes #38

#### Changes proposed in this Pull Request:
- Remove `tslint.json` when pxblue `.eslintrc` is present

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,15 @@
 # Changelog
 
+## v1.0.5
```
Comment (joebochill) [inline CHANGELOG.md:3]: Just so we don't run into the case we had before where we added multiple new versions in the changelog even though none had actually been published yet and they could be combined:
```suggestion
## v1.0.5 (Not Released)
```

File: __tests__/cli-integration.test.ts [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-const cli = async cmd =>
-  system.run('node ' + filesystem.path(src, 'bin', 'pxb') + ` ${cmd}`)
+const cli = async (cmd) => system.run('node ' + filesystem.path(src, 'bin', 'pxb') + ` ${cmd}`);
 
 test('outputs version', async () => {
-  const output = await cli('--version')
-  expect(output).toContain('1.0.4')
-})
+    const output = await cli('--version');
+    expect(output).toContain(packageJSON.version);
```
Comment (joebochill) [inline __tests__/cli-integration.test.ts:10]: Haha, nice! This test is silly and it tripped us up every time.

**Comment by emclaug2** [comment]
How hard would it be to remove the "tslint" entry in the angular.json file?

It's not hurting anything, it's just not required now that the tslint file is gone.

**Comment by emclaug2** [comment]
> How hard would it be to remove the "tslint" entry in the angular.json file?
> 
> It's not hurting anything, it's just not required now that the tslint file is gone.

Same goes with tslint dependency in package.json.

**Comment by joebochill** [comment]
Modifying angular.json should be pretty easy since we can read it in as a JSON object and modify it as we see fit. And to remove dependencies, we can use `yarn remove` or `npm uninstall` to clear them from the package.json file and the node_modules.

File: src/extensions/file-extensions.ts [hunk 3]
```diff
@@ -38,11 +38,38 @@ module.exports = (toolbox: GluegunToolbox): void => {
...
+            }
+            print.info(output);
+
+            // remove lint attribute in angular.json
+            if (filesystem.exists(pathInFolder('angular.json'))) {
+                const angularJSON = filesystem.read(pathInFolder('angular.json'), 'json');
+                angularJSON.projects[projectName].architect.lint = undefined;
+                filesystem.write(pathInFolder('angular.json'), JSON.stringify(angularJSON, null, 4));
+            }
+        }
```
Comment (joebochill) [inline src/extensions/file-extensions.ts:71]: I'm leaning towards having this logic in the pxblue-extensions file. This function is a shared function for 'adding lint config', but now it's adding lint configs and also conditionally removing lint configs.

Might make more sense to just put another blurb in the addPXBlueAngular function after adding the lint dependencies that does 'clean up tslint dependencies'

Comment (joebochill) [reply src/extensions/file-extensions.ts:71]: You'll have more access to yarn vs. npm and the fact that you know the files exist if it's in the other extension.

Comment (joebochill) [reply src/extensions/file-extensions.ts:71]: Are we able (safely) to remove the lint line, or is setting it to undefined the way to do it?

Comment (huayunh) [reply src/extensions/file-extensions.ts:71]: > Are we able (safely) to remove the lint line

Shouldn't make any difference, but I am changing it to using the `delete` keyword instead.

**Pull request merged**